core/ArchiveProcessor/Rules.php

```diff
@@ -343,4 +344,53 @@ public static function shouldProcessSegmentsWhenReArchivingReports()
     {
         return Config::getInstance()->General['rearchive_reports_in_past_exclude_segments'] != 1;
     }
+
+    public static function shouldForceArchiveInvalidatedArchiveInThisRequest(Parameters $params, $doneFlags)
+    {
+        $isArchivePhpTrigerred = !SettingsServer::isArchivePhpTriggered(); // safety to make sure core:archive doesn't trigger archiving if the archive is still usable
+        if (!$isArchivePhpTrigerred) {
+            return false;
+        }
+
+        $isArchivingEnabledFor = !Rules::isArchivingDisabledFor([$params->getSite()->getId()], $params->getSegment(), $params->getPeriod()->getLabel());
```

Comment:
fyi same here the variable `isArchivingEnabledFor` is actually the opposite? Be great to maybe remove the `!` and then also in the `if` below remove the `!` then it be easier to read

Reply:
the variable is `Enabled`, but the function is `!Disabled`, so this is correct